templates/geosafe/analysis/dynamic_scripts/create/transparency_control.html

```diff
@@ -152,6 +152,31 @@
                 $("#impact-legend-img").attr("src", legend_url);
                 impact_layer.setZIndex(3);
             }
+
+            // Zooming magic
+            // If only one of the 3 layers is selected then zoom to the layer
+            if ((is_hazard != is_exposure) != is_impact) {
+              map.fitBounds([[layer_bbox_y1, layer_bbox_x1], [layer_bbox_y0, layer_bbox_x0]]);
+            }
+            // If hazard and exposure selected intersect both extents
+            if ((layer_purpose=='hazard' && is_exposure) || (layer_purpose=='exposure' && is_hazard)) {
```

Comment:
I moved this zooming part after all the boolean variables (is_hazard, is_exposure, and is_impact are updated to reflect what the active/selected layers are). Intersecting happens when user selects hazard (by layer_purpose) and the exposure is active OR when user select exposure and the hazard layer is active